web/client-api/src/main/java/io/deephaven/web/client/api/i18n/JsTimeZone.java

```diff
@@ -11,123 +11,555 @@
 
 @JsType(name = "TimeZone", namespace = "dh.i18n")
 public class JsTimeZone {
-    // Cache the time zones that are parsed so we don't need to recalculate them
-    private static final Map<String, JsTimeZone> timeZoneCache = new HashMap<>();
+    // Map of all time IDs to their JsTimeZone
+    private static final Map<String, JsTimeZone> timeZones = new HashMap<>();
 
     public static JsTimeZone getTimeZone(String tzCode) {
-        return timeZoneCache.computeIfAbsent(tzCode, ignored -> createTimeZone(tzCode));
+        return getTimeZoneForCode(tzCode);
     }
 
-    private static JsTimeZone createTimeZone(String tzCode) {
-        if (tzCode.equals("UTC") || tzCode.equals("GMT") || tzCode.equals("Etc/GMT") || tzCode.equals("Z")) {
-            return new JsTimeZone(TimeZone.createTimeZone(0));
+    private static JsTimeZone getTimeZoneForCode(String tzCode) {
+        initTimeZones();
+
+        if (!timeZones.containsKey(tzCode)) {
+            throw new IllegalArgumentException("Unsupported time zone " + tzCode);
+        }
+
+        return timeZones.get(tzCode);
+    }
+
+    private static void initTimeZones() {
```

Comment:
this could be put into a `static {...}` block, then the language will ensure it is only called once, and that will be the first time this class is used.